packages/ocaml-config/ocaml-config.2/files/gen_dkml_config.ml.in

```diff
@@ -0,0 +1,66 @@
+let () =
+  let ocaml_version =
+    let v = Sys.ocaml_version in
+    let l = String.length v in
+    let plus = try String.index v '+' with Not_found -> l in
+    (* Introduced in 4.11.0; used from 4.12.0 *)
+    let tilde = try String.index v '~' with Not_found -> l in
+    String.sub v 0 (min (min plus tilde) l)
+  in
+  if ocaml_version <> Sys.argv.(1) then
+    (Printf.eprintf
+       "OCaml version mismatch: %%s, expected %s"
+       ocaml_version Sys.argv.(1);
+     exit 1)
+  else
+  let oc = open_out (Sys.argv.(2) ^ ".config") in
+  let exe = ".exe" in
+  let (ocaml, suffix) =
+    let s = Sys.executable_name in
+    if Filename.check_suffix s exe then
+      (Filename.chop_suffix s exe, exe)
+    else
+      (s, "")
+  in
+  let ocamlc = ocaml^"c"^suffix in
+  let libdir =
+    if Sys.command (ocamlc^" -where > where") = 0 then
+      (* Must be opened in text mode for Windows *)
+      let ic = open_in "where" in
+      let r = input_line ic in
+      close_in ic; r
+    else
+      failwith "Bad return from 'ocamlc -where'"
+  in
+  let stubsdir =
+    let ic = open_in (Filename.concat libdir "ld.conf") in
+    let rec r acc = try r (input_line ic::acc) with End_of_file -> acc in
+    let lines = List.rev (r []) in
+    close_in ic;
+    (* This needs to be corrected!
+       The original version in Opam repository assumes Unix with:
+         String.concat ":" lines
+       opam-repository-mingw corrects it.
+     *)
+    let sep = if Sys.os_type = "Win32" then ";" else ":" in
+    String.concat sep lines
+  in
+  let p fmt = Printf.fprintf oc (fmt ^^ "\n") in
+  p "opam-version: \"2.0\"";
+  p "variables {";
+  p "  native: %%b"
+    (Sys.file_exists (ocaml^"opt"^suffix));
+  (* This needs to be corrected!
+     The original version in Opam repository checked for (ocamlc^".opt"^suffix) but
+     ocamlc already has a suffix on Windows (ex. '...\bin\ocamlc.exe').
+   *)
+  p "  native-tools: %%b"
+    (Sys.file_exists (ocaml^"c.opt"^suffix));
+  p "  native-dynlink: %%b"
+    (Sys.file_exists (Filename.concat libdir "dynlink.cmxa"));
+  p "  stubsdir: %%S"
+    stubsdir;
+  p "  preinstalled: %{ocaml-system:installed}%";
+  p "  compiler: \"%{ocaml-system:installed?system:}%%{dkml-base-compiler:version}%%{dkml-variants:version}%%{dkml-option-32bit:installed?+32bit:}%%{dkml-option-afl:installed?+afl:}%%{dkml-option-bytecode-only:installed?+bytecode-only:}%%{dkml-option-default-unsafe-string:installed?+default-unsafe-string:}%%{dkml-option-fp:installed?+fp:}%%{dkml-option-flambda:installed?+flambda:}%%{dkml-option-musl:installed?+musl:}%%{dkml-option-nnp:installed?+nnp:}%%{dkml-option-no-flat-float-array:installed?+no-flat-float-array:}%%{dkml-option-spacetime:installed?+spacetime:}%%{dkml-option-static:installed?+static:}%\"";
```

Comment:
The `dkml-base-compiler:version` part is obviously needed - I hope it can be OK to eliminate dkml-option packages and use the ocaml-option ones instead? It's absolutely fine for the `available`, `conflicts` and `depends` of those packages to be edited to support `dkml-base-compiler` - from the opam-repository user's perspective, I don't think having two sets of these packages is a good idea.

Reply:
Fixed in the other PR